### PR TITLE
Fix index bug for TextFeaturizer and LSA

### LINKED ISSUE #1643: Issue running AutoMLSearch with text columns and LSA
```
from sklearn import datasets
from evalml.automl.callbacks import raise_error_callback
X, y = datasets.make_classification(n_samples=12, n_features=10,
                                        n_informative=2, n_redundant=2, random_state=0)
X = pd.DataFrame(X)
X['text'] = pd.Series(['hey this is supposed to be a text column, hey this is supposed to be a text column because these strings are super long',
                       "he helped her walk,", "me me me eat food", "the sentence doth long", "this is another text string",
                      "help oh help", "this is crazy", "happy new years", "wow so fun this is",
                      "wowowowow", "evalml is cool", "yay this is it"])
y = pd.Series([0] * 6 + [1] * 6)
automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', max_iterations=10, error_callback=raise_error_callback)
automl.search()
```

raises:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-7-f3a9dafe1b51> in <module>
      2 
      3 automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', max_iterations=10, error_callback=raise_error_callback)
----> 4 automl.search()

~/Desktop/evalml/evalml/automl/automl_search.py in search(self, data_checks, show_iteration_plot)
    515 
    516             current_batch_size = len(current_batch_pipelines)
--> 517             current_batch_pipeline_scores = self._evaluate_pipelines(current_batch_pipelines, search_iteration_plot=search_iteration_plot)
    518 
    519             # Different size indicates early stopping

~/Desktop/evalml/evalml/automl/automl_search.py in _evaluate_pipelines(self, current_pipeline_batch, baseline, search_iteration_plot)
    784                                     self._start, 1 if baseline else self._automl_algorithm.batch_number, self.show_batch_output)
    785 
--> 786                 evaluation_results = self._compute_cv_scores(pipeline)
    787                 parameters = pipeline.parameters
    788 

~/Desktop/evalml/evalml/automl/automl_search.py in _compute_cv_scores(self, pipeline)
    689                 if self.error_callback is not None:
    690                     self.error_callback(exception=e, traceback=traceback.format_tb(sys.exc_info()[2]), automl=self,
--> 691                                         fold_num=i, pipeline=pipeline)
    692                 if isinstance(e, PipelineScoreError):
    693                     nan_scores = {objective: np.nan for objective in e.exceptions}

~/Desktop/evalml/evalml/automl/callbacks.py in raise_error_callback(exception, traceback, automl, **kwargs)
     13     logger.error(f'AutoMLSearch raised a fatal exception: {str(exception)}')
     14     logger.error("\n".join(traceback))
---> 15     raise exception
     16 
     17 

~/Desktop/evalml/evalml/automl/automl_search.py in _compute_cv_scores(self, pipeline)
    677                 cv_pipeline = pipeline.clone(pipeline.random_state)
    678                 logger.debug(f"\t\t\tFold {i}: starting training")
--> 679                 cv_pipeline.fit(X_train, y_train)
    680                 logger.debug(f"\t\t\tFold {i}: finished training")
    681                 cv_pipeline = self._tune_binary_threshold(cv_pipeline, X_threshold_tuning, y_threshold_tuning)

~/Desktop/evalml/evalml/utils/base_meta.py in _set_fit(self, X, y)
     16         @wraps(method)
     17         def _set_fit(self, X, y=None):
---> 18             return_value = method(self, X, y)
     19             self._is_fitted = True
     20             return return_value

~/Desktop/evalml/evalml/pipelines/classification_pipeline.py in fit(self, X, y)
     45         self._encoder.fit(y)
     46         y = self._encode_targets(y)
---> 47         self._fit(X, y)
     48         return self
     49 

~/Desktop/evalml/evalml/pipelines/pipeline_base.py in _fit(self, X, y)
    195 
    196     def _fit(self, X, y):
--> 197         self._component_graph.fit(X, y)
    198         self.input_feature_names = self._component_graph.input_feature_names
    199 

~/Desktop/evalml/evalml/pipelines/component_graph.py in fit(self, X, y)
     87             y (pd.Series): The target training data of length [n_samples]
     88         """
---> 89         self._compute_features(self.compute_order, X, y, fit=True)
     90         return self
     91 

~/Desktop/evalml/evalml/pipelines/component_graph.py in _compute_features(self, component_list, X, y, fit)
    199             else:
    200                 if fit:
--> 201                     component_instance.fit(input_x, input_y)
    202                 if not (fit and component_name == self.compute_order[-1]):  # Don't call predict on the final component during fit
    203                     output = component_instance.predict(input_x)

~/Desktop/evalml/evalml/utils/base_meta.py in _set_fit(self, X, y)
     16         @wraps(method)
     17         def _set_fit(self, X, y=None):
---> 18             return_value = method(self, X, y)
     19             self._is_fitted = True
     20             return return_value

~/Desktop/evalml/evalml/pipelines/components/component_base.py in fit(self, X, y)
    100         try:
    101             import pdb; pdb.set_trace()
--> 102             self._component_obj.fit(X, y)
    103             return self
    104         except AttributeError:

~/Desktop/venv/lib/python3.7/site-packages/sklearn/linear_model/_logistic.py in fit(self, X, y, sample_weight)
   1342         X, y = self._validate_data(X, y, accept_sparse='csr', dtype=_dtype,
   1343                                    order="C",
-> 1344                                    accept_large_sparse=solver != 'liblinear')
   1345         check_classification_targets(y)
   1346         self.classes_ = np.unique(y)

~/Desktop/venv/lib/python3.7/site-packages/sklearn/base.py in _validate_data(self, X, y, reset, validate_separately, **check_params)
    430                 y = check_array(y, **check_y_params)
    431             else:
--> 432                 X, y = check_X_y(X, y, **check_params)
    433             out = X, y
    434 

~/Desktop/venv/lib/python3.7/site-packages/sklearn/utils/validation.py in inner_f(*args, **kwargs)
     70                           FutureWarning)
     71         kwargs.update({k: arg for k, arg in zip(sig.parameters, args)})
---> 72         return f(**kwargs)
     73     return inner_f
     74 

~/Desktop/venv/lib/python3.7/site-packages/sklearn/utils/validation.py in check_X_y(X, y, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, multi_output, ensure_min_samples, ensure_min_features, y_numeric, estimator)
    800                     ensure_min_samples=ensure_min_samples,
    801                     ensure_min_features=ensure_min_features,
--> 802                     estimator=estimator)
    803     if multi_output:
    804         y = check_array(y, accept_sparse='csr', force_all_finite=True,

~/Desktop/venv/lib/python3.7/site-packages/sklearn/utils/validation.py in inner_f(*args, **kwargs)
     70                           FutureWarning)
     71         kwargs.update({k: arg for k, arg in zip(sig.parameters, args)})
---> 72         return f(**kwargs)
     73     return inner_f
     74 

~/Desktop/venv/lib/python3.7/site-packages/sklearn/utils/validation.py in check_array(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, estimator)
    643         if force_all_finite:
    644             _assert_all_finite(array,
--> 645                                allow_nan=force_all_finite == 'allow-nan')
    646 
    647     if ensure_min_samples > 0:

~/Desktop/venv/lib/python3.7/site-packages/sklearn/utils/validation.py in _assert_all_finite(X, allow_nan, msg_dtype)
     97                     msg_err.format
     98                     (type_err,
---> 99                      msg_dtype if msg_dtype is not None else X.dtype)
    100             )
    101     # for object dtype data, we only check for NaNs (GH-13254)

ValueError: Input contains NaN, infinity or a value too large for dtype('float64').
```

I think this has to do with the indices: prior to the text component, some indices are dropped. When transformation happens with the original, undropped indices, NaN values are filled in for those dropped rows, and passed to the estimator which throws this.

If I'm doing something extremely wrong here, please let me know! It's curious to me that this isn't failing in docs/tests.

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #1643 

Fixes an index bug where if the original input DF has custom indices, NaNs are backfilled. In the original repro, we don't specify indices specifically but automl passes in indices split by our data split.

### CODE REVIEW

File: evalml/tests/conftest.py [hunk 1]
```diff
@@ -158,6 +158,19 @@ def X_y_categorical_classification():
     return X, y
 
 
+@pytest.fixture()
```
Comment (angela97lin) [inline evalml/tests/conftest.py:161]: Moved this here, since LSA and TextFeaturizer both use it :)

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -4,16 +4,17 @@ Release Notes
 **Future Releases**
     * Enhancements
         * Added time series support for ``make_pipeline`` :pr:`1566`
-        * Added multiclass check to InvalidTargetDataCheck for two examples per class :pr:`1596`
+        * Added multiclass check to ``InvalidTargetDataCheck`` for two examples per class :pr:`1596`
     * Fixes
-        * Fix thresholding for pipelines in AutoMLSearch to only threshold binary classification pipelines :pr:`1622` :pr:`1626`
+        * Fixed thresholding for pipelines in ``AutoMLSearch`` to only threshold binary classification pipelines :pr:`1622` :pr:`1626`
```
Comment (angela97lin) [inline docs/source/release_notes.rst:9]: Just standardizing :d

File: evalml/tests/conftest.py (hunk 1 above)
Comment (ParthivNaresh) [reply evalml/tests/conftest.py:161]: Good call!

**Review by bchen1116 — APPROVED** [review]
nice!

**Review by freddyaboulton — APPROVED** [review]
Thanks for the fix @angela97lin !

@angela97lin and I tried to figure out why we're running into this problem now for the first time (the repro on the ticket is just a vanilla use-case of automl with text). We noticed this must have been introduced since the 0.17.0 release since the [text demo on latest](https://evalml.alteryx.com/en/latest/demos/text_input.html) is "failing" but it's "passing" on [stable](https://evalml.alteryx.com/en/stable/demos/text_input.html).

Nothing in our release notes is related to the text featurizer so maybe this has something to do with the latest featuretools release (released two days after 0.17.0). That being said, this fixes the docs locally so let's get it merged!

**Review by chukarsten — APPROVED** [review]

**Pull request merged**